src/api/libcellml/componententity.h

```diff
@@ -329,12 +321,11 @@ class LIBCELLML_EXPORT ComponentEntity: public ImportedEntity
      * @brief Take the component at the given @p index and return it.
      *
      * Removes the component at the given @p index position and returns it.
-     * If an invalid @p index is passed to the method a @c std::out_of_range
-     * exception is thrown.
+     * @p index must be in the range [0, #components).
      *
-     * @param index The index of the Component to take (zero-based).
+     * @param index The index of the Component to take.
```

Comment:
Minor: if "(zero-based)" is to be removed, then it ought to be removed elsewhere in the code too.